### DISCUSSION TITLE
Add solutions for exercises

### DISCUSSION THREAD

**Discussion by wrongbyte** [browserengineering/book#289] [Ideas]
I am currently following the code for the chapter 1, and I found the exercises part very interesting.
However, there are things - for example dealing with chunked data and gzip encoding - that present different behaviors depending on the URL to which I make the request. Although it may be a problem with my code, it would be very helpful if there were:
1 - A part in this repository containing the resolution of the exercises, for reference
2 - Examples of behaviors expected depending on URL requested, or something like that. Just like http://browser.engineering/redirect makes your browser redirects to another URL.

**Comment by pavpanchekha** [comment]
Hi @wrongbyte, we'd like to do more of this. Solutions are unlikely for now because we also use the exercises to teach University courses and don't want solutions easily accessible to students, but we've been working on putting together test suites for each exercise. Some exercises are trickier than others—I think the Transfer-Encoding and gzip exercise is particularly variable across servers.

**Reply by Zhencha0Cai** [reply]
hi @pavpanchekha, are these test suites for each exercise available now? The lab*-test.md file doesn't have test cases for exercises.

**Reply by pavpanchekha** [reply]
Not every exercise, but for many of them yes! I keep them here, in my teaching repo: https://github.com/cs4560-utah-sp24/test

**Reply by Zhencha0Cai** [reply]
> Not every exercise, but for many of them yes! I keep them here, in my teaching repo: https://github.com/cs4560-utah-sp24/test

Thank you very much!
